src/NUnitFramework/tests/Attributes/TestCaseAttributeTests.cs

```diff
@@ -748,5 +749,38 @@ public async Task<T> TestWithAsyncGenericReturnType<T>(T arg1)
         }
 
         #endregion
+
+        [TestCase(TypeArgs = new[] { typeof(int) })]
+        public void ExplicitTypeArgsWithoutParameters<T>()
+        {
+            Assert.That(typeof(T), Is.EqualTo(typeof(int)));
+        }
+
+        [TestCase(2, TypeArgs = new[] { typeof(int) })]
```

Comment:
Doesn't this already without having to add `TypeArgs` as the `TryGetTypeArguments` determines the types from the actual parameters?